crates/forge-runner/src/lib.rs

```diff
@@ -236,17 +283,22 @@ pub async fn run_tests_from_crate(
             continue;
         };
 
-        let function = runner.find_function(&case_name)?;
+        let function = sierra_program
+            .funcs
+            .iter()
+            .find(|f| f.id.debug_name.clone().unwrap().ends_with(&case_name))
+            .unwrap();
```

Comment:
So we still kind of rely on sierra here don't we?

Reply:
We do, but this could be replaced. What we need is offsets to the function and it's argument and return types, and this information cannot be obtained from the casm code. My idea was that whatever module will be responsible for the compilation from sierra to casm, it will also output these information in some format.

Reply:
Let's make a TODO for that perhaps, and discuss on the friday's meeting about compilation from sierra to casm

Reply:
Added TODOs in code. Not creating issues for these for now